db/migrate/20160112133345_add_type_to_lookup_value.rb

```diff
@@ -0,0 +1,11 @@
+class AddTypeToLookupValue < ActiveRecord::Migration
+  def change
+    add_column :lookup_values, :type, :string
+    LookupValue.reset_column_information
+    LookupValue.all.each do |lookup_value|
+      subclass = lookup_value.lookup_key.puppet? ?  'PuppetLookupValue' : 'LookupValue'
+      lookup_value.type = subclass
+      lookup_value.save
```

Comment:
Avoid looping over every single LookupValue and saving it, that's expensive and relies on using the model.  Preferably use a simple update against all lookup values, which should be possible if you join to the lookup_keys table to identify which rows to updaet.